azurelinuxagent/ga/update.py

```diff
@@ -614,6 +614,10 @@ def _process_goal_state(self, exthandlers_handler, remote_access_handler):
                 self._extensions_summary = ExtensionsSummary()
                 exthandlers_handler.run()
 
+                # check cgroup and disable if any extension started in agent cgroup after goal state processed.
+                # Note: Monitor thread periodically checks this in addition to here.
+                CGroupConfigurator.get_instance().check_cgroups()
```

Comment:
since now this is called from 2 threads, please verify the method is thread-safe or add a lock